### PR TITLE
GEM-44 | Add post for group attachment

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+@RestController
+@RequestMapping("/external/attachments/groups")
+class GroupController(
+    val groupService: GroupService,
+) {
+
+    @PostMapping("/{groupId}", consumes = [APPLICATION_OCTET_STREAM_VALUE], produces = [APPLICATION_JSON_INTERNAL_VER_1])
+    @ResponseStatus(CREATED)
+    fun createProduct(
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt:24]: wrong name

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        whenever(fileDetector.getFileSize(data)).thenReturn(attachment.sizeInBytes)
+        whenever(fileDetector.getFileMediaType(data)).thenReturn(attachment.contentType)
+        whenever(groupAttachmentRepository.save(any())).thenReturn(attachment)
+
+        // when
+        val groupAttachment = groupService.saveAttachment(data, attachment.groupId, attachment.uploadedByUser)
+
+        // then
+        verify(groupAttachmentRepository, times(1)).save(any())
+        groupAttachment.also {
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt:46]: Can't we just write it like that: ``` groupAttachment shouldBe attachment``` ?

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        whenever(groupAttachmentRepository.save(any())).thenReturn(attachment)
+
+        // When & Then
+        shouldThrow<AttachmentContentTypeNotSupportedException> {
+            groupService.saveAttachment(data, attachment.groupId, attachment.uploadedByUser)
+        }
+        verify(groupAttachmentRepository, times(0)).save(any())
+    }
+
+    should("throw an exception for unsupported content type") {
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]:
```suggestion
    should("throw AttachmentSizeExceededException when attachment size is to big") {
```

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        // given
+        val data = LARGE_FILE
+
+        // when % then
+        shouldThrow<AttachmentSizeExceededException> {
+            fileDetector.getFileSize(data)
+        }
+    }
+
+    should("detect correct size of file") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]:
```suggestion
    should("detect correct media type of file") {
```

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        // given
+        val data = SMALL_FILE
+
+        // when
+        val dataSize = fileDetector.getFileMediaType(data)
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]:
```suggestion
        val mediaType = fileDetector.getFileMediaType(data)
```

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+        // then
+        dataSize shouldBe IMAGE_JPEG_VALUE
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]:
```suggestion
        mediaType shouldBe IMAGE_JPEG_VALUE
```

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    should("throw AttachmentContentTypeNotSupportedException when size is exceeded") {
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]:
```suggestion
    should("throw AttachmentContentTypeNotSupportedException when content type is not supported") {
```

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 2 above)
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: Can't we just write it like that: ``` groupAttachment shouldBe attachment``` ?

File: src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt (hunk 1 above)
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt]: wrong name

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 2 above)
Comment (lawos98) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt:46]: for me its easy to modify one place that 12

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 3 above)
Comment (lawos98) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 4 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 5 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt:55]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 6 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 7 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt:61]: changed

File: src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt (hunk 1 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/controller/GroupController.kt]: changed

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 2 above)
Comment (lawos98) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 7 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 6 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 5 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt (hunk 4 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/detector/TikaFileDetectorIT.kt]: changed

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 3 above)
Comment (lawos98) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: changed

File: src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt (hunk 2 above)
Comment (pwgacek) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: you didn't

Comment (lawos98) [reply src/test/kotlin/pl/edu/agh/gem/internal/service/GroupServiceTest.kt]: true changed i an other place also

**Review by pwgacek — APPROVED** [review]

**Pull request merged**